### ISSUE TITLE
[Phase 3] Model Implementation - CNN Architecture

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/protein-structure-prediction#4]
## Objective

Implement 1D CNN architecture for per-residue secondary structure classification.

**Part of:** #1

## Dependencies

- #2 (Data Infrastructure) complete
- #3 (Feature Engineering) complete

### Base Model Interface
- [ ] Create `src/models/base.py` with abstract base class
- [ ] Define fit, predict, save, load interface

### CNN Architecture
- [ ] Create `src/models/cnn.py` with `ProteinCNN`
- [ ] Implement 1D convolutions for local patterns
- [ ] Add per-residue classification head

### Training Loop
- [ ] Create `src/models/trainer.py` with `ModelTrainer`
- [ ] Implement early stopping
- [ ] Add learning rate scheduling

### Configuration
- [ ] Create `configs/model_configs.yaml`

### CNN Architecture

```python
# src/models/cnn.py
import torch
import torch.nn as nn

class ProteinCNN(nn.Module):
    def __init__(self, input_dim=21, hidden_dim=128, num_classes=3):
        super().__init__()
        self.conv1 = nn.Conv1d(input_dim, hidden_dim, kernel_size=7, padding=3)
        self.conv2 = nn.Conv1d(hidden_dim, hidden_dim, kernel_size=5, padding=2)
        self.conv3 = nn.Conv1d(hidden_dim, num_classes, kernel_size=3, padding=1)
        self.relu = nn.ReLU()
        self.dropout = nn.Dropout(0.3)

    def forward(self, x):
        # x: (batch, channels, length)
        x = self.relu(self.conv1(x))
        x = self.dropout(x)
        x = self.relu(self.conv2(x))
        x = self.dropout(x)
        x = self.conv3(x)  # (batch, 3, length)
        return x
```

### Training Loop

```python
# src/models/trainer.py
import torch
from torch.optim import Adam
from torch.nn import CrossEntropyLoss

class ModelTrainer:
    def __init__(self, model, lr=0.001):
        self.model = model
        self.optimizer = Adam(model.parameters(), lr=lr)
        self.criterion = CrossEntropyLoss()

    def train_epoch(self, dataloader):
        self.model.train()
        total_loss = 0
        for batch_x, batch_y in dataloader:
            self.optimizer.zero_grad()
            logits = self.model(batch_x)
            loss = self.criterion(logits, batch_y)
            loss.backward()
            self.optimizer.step()
            total_loss += loss.item()
        return total_loss / len(dataloader)
```

## Definition of Done

- [ ] CNN achieves Q3 > 65% on CB513 test set
- [ ] Training completes in <30 minutes for 50 epochs
- [ ] Model saves and loads correctly
- [ ] Loss decreases during training

## Technical Notes

- Use per-residue CrossEntropyLoss
- Class weights may help with imbalanced classes
- Kernel sizes 3-7 capture local structural patterns
